src/plots/ternary/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Ternary = require('./ternary');
+
+var Plots = require('../../plots/plots');
+
+
+exports.name = 'ternary';
+
+exports.attr = 'ternary';
+
+exports.idRoot = 'ternary';
+
+exports.idRegex = /^ternary([2-9]|[1-9][0-9]+)?$/;
+
+exports.attrRegex = /^ternary([2-9]|[1-9][0-9]+)?$/;
```

Comment:
non- :no_entry_sign:  , but we should put those `([2-9]|[1-9][0-9]+)?$` in a file in `src/constants/`